components/modal/modal.component.ts

```diff
@@ -76,8 +75,7 @@ export class ModalDirective implements AfterViewInit, OnDestroy {
 
   @HostListener('click', ['$event'])
   protected onClick(event:any):void {
-    if (this.config.backdrop !== 'static' &&
-        event.target === this.element.nativeElement) {
+    if (this.config.ignoreBackdropClick || this.config.backdrop === 'static') {
```

Comment:
Hi Dimi,

You are absolutely right. it should be reversed. I will do it.

Thanks,
Pradeep

On Mon, Jul 11, 2016 at 3:31 PM, Dmitriy Shekhovtsov <
notifications@github.com> wrote:

> In components/modal/modal.component.ts
> <https://github.com/valor-software/ng2-bootstrap/pull/708#discussion_r70231022>
> :
>
> > @@ -76,8 +75,7 @@ export class ModalDirective implements AfterViewInit, OnDestroy {
> >
> >    @HostListener('click', ['$event'])
> >    protected onClick(event:any):void {
> > -    if (this.config.backdrop !== 'static' &&
> > -        event.target === this.element.nativeElement) {
> > +    if (this.config.ignoreBackdropClick || this.config.backdrop === 'static') {
>
> @pradeepsekar <https://github.com/pradeepsekar> seem this logic should be
> reversed?
> if ignoreBackdropClick is true or this.config.backdrop === 'static' modal
> should not be hidden
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/valor-software/ng2-bootstrap/pull/708/files/6993f2147fee98399971864b7039a790bcee75ae#r70231022>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACouvHKNK6m_EOcelnKA9hBWkqSVV-Hqks5qUhRggaJpZM4JIv4_>
> .
>